### DISCUSSION TITLE
Site to site IPsec to Fortinet

### DISCUSSION THREAD

**Discussion by binojvs** [strongswan/strongswan#499] [General]
Dear All,

I am new to the Strongswan concept. We are trying to enable a site to site connection to a fortigate unit.

Tunnel is not coming up. 

Fortinet side configuration as follows.

 config vpn ipsec phase1-interface
    edit "Twixor"
        set interface "LL-VPN"
        set ike-version 2
        set peertype any
        set net-device disable
        set proposal aes128-sha256
        set dhgrp 2
        set remote-gw 178.128.16.**
        set psksecret ENC wZdV6BmW0kH7VSoIEd9TAz1WwcMp53CuEJD8SGwo4zTEzg6jkwgTl+TBXvEjt1hB2Dfk0BmOKzeJJFuUvSW8aKQhbZ4/xeUqdbwjDoTuUhfazMMHF3fsVbpFyj9zpADzRvRsFJY6KFnvj0ovzQo9Q==
    next
end

config vpn ipsec phase2-interface
    edit "Twixor Network"
        set phase1name "Twixor"
        set proposal aes128-sha256
        set pfs disable
        set auto-negotiate enable
        set src-subnet 172.16.152.116 255.255.255.255
        set dst-subnet 10.130.0.0 255.255.0.0
    next
end

Ubuntu Side 

IPsec.conf

config setup
    charondebug="ike 4, knl 2, cfg 2 net 2"
    #charondebug = "all"
    uniqueids=yes

conn ikev2-vpn-vps
    auto=route
    type=tunnel
    keyexchange=ikev2
    forceencaps=yes
    ike=aes128-sha256-modp1024
    esp=aes128-sha256-modp1024
    dpdaction=clear
    dpddelay=300s
    left=178.128.16.**
    leftid=vpsvpn
    leftsubnet=10.130.0.0/16
    right=151.253.125.**
    rightid=151.253.125.**
    rightdns=8.8.8.8,8.8.4.4
    rightsourceip=172.16.152.116/32
    authby=psk
    leftauth=psk
    rightauth=psk
    pfs=no
    ikelifetime=24h
    lifetime=12h
    eap_identity=%identity

tcpdump -nnnnvvv -i eth0 host 151.253.125.**
tcpdump: listening on eth0, link-type EN10MB (Ethernet), capture size 262144 bytes
06:40:57.079758 IP (tos 0x0, ttl 54, id 51689, offset 0, flags [none], proto UDP (17), length 340)
    151.253.125.**.500 > 178.128.16.**.500: [udp sum ok] isakmp 2.0 msgid 00000000 cookie 2d6a2e13e88a4916->0000000000000000: parent_sa ikev2_init[I]:
    (sa: len=44
        (p: #1 protoid=isakmp transform=4 len=44
            (t: #1 type=encr id=aes (type=keylen value=0080))
            (t: #2 type=prf id=#5 )
            (t: #3 type=integ id=#12 )
            (t: #4 type=dh id=modp1024 )))
    (v2ke: len=128 group=modp1024 519def9cc03d0b288b7dd528c5594c967cf6148acfc1208b2f328915b80d220b85b72968518fc03c1b2f158a416577a0510cbcf52ab58511f5137e3b8914f5f49ac13d0f63461c027f3b8444a8177a6825db0e75382ebdffdee8a434695bfe6474918b7ef88acd0d81a460d838ab224540eb22aa52e4f67b7c5957e0a664096b)
    (nonce: len=32 nonce=(7d1c8776ed699a95ddab14f9923d128dabdc129827d491a6f720da1035bca830) )
    (n: prot_id=#0 type=16388(nat_detection_source_ip))
    (n: prot_id=#0 type=16389(nat_detection_destination_ip))
    (n: prot_id=#0 type=16430(status))
06:40:57.080075 IP (tos 0x0, ttl 64, id 13735, offset 0, flags [none], proto UDP (17), length 365)
    178.128.16.**.500 > 151.253.125.**.500: [udp sum ok] isakmp 2.0 msgid 00000000 cookie 2d6a2e13e88a4916->755996a7539e5d26: parent_sa ikev2_init[R]:
    (sa: len=44
        (p: #1 protoid=isakmp transform=4 len=44
            (t: #1 type=encr id=aes (type=keylen value=0080))
            (t: #2 type=integ id=#12 )
            (t: #3 type=prf id=#5 )
            (t: #4 type=dh id=modp1024 )))
    (v2ke: len=128 group=modp1024 ca8e4449536b4d7f86f7f9633415405a7c4b3ee046838cd7130bc0294cb8fe586cb4ed4fa4699ceac4ff3f4725acbcfe4f6ab1b58be4e9c8dbff4b09f8163316690bbf4a406b9b75ecd647d7b16b6d80a87fce9ab96dae68f5cf59cec0af3690caeab479de8fab02f47c6f825a9e200b3a9f49ce3c334aca1221fae24968d388)
    (nonce: len=32 nonce=(20d49fee7cfa0c9ceee90a72a2200755201440a42f1a43499b0f8b1f78fbfc6a) )
    (n: prot_id=#0 type=16388(nat_detection_source_ip))
    (n: prot_id=#0 type=16389(nat_detection_destination_ip))
    (v2cr: len=21 04dfc55921c5256f00589bd2b3cd690f6b159bba27)
    (n: prot_id=#0 type=16404(status))
06:41:06.131577 IP (tos 0x0, ttl 54, id 51690, offset 0, flags [none], proto UDP (17), length 340)
    151.253.125.**.500 > 178.128.16.**.500: [udp sum ok] isakmp 2.0 msgid 00000000 cookie ca1cc0c13da41457->0000000000000000: parent_sa ikev2_init[I]:
    (sa: len=44
        (p: #1 protoid=isakmp transform=4 len=44
            (t: #1 type=encr id=aes (type=keylen value=0080))
            (t: #2 type=prf id=#5 )
            (t: #3 type=integ id=#12 )
            (t: #4 type=dh id=modp1024 )))
    (v2ke: len=128 group=modp1024 b32ae7b10cf8d0557b0914fc3241fcfbd3b26c4662b6513402d89dd20cd95033d9fe4ebf7ca954668a217a4399c1fef482ed6aaaab6aa63a08e447dbc27df8e50395a16a474b151998a78b0c2449c5bb4522c5ca19bd8622f2d71641fcadaa6bce87a5716d074c593d43767f05cb3fe59e13eb1d6e5b291ce03ca26450dabe81)
    (nonce: len=32 nonce=(7d9bb11894b4c528866e043fd7dc39d483ed8889280132a99ae7a247b43584a5) )
    (n: prot_id=#0 type=16388(nat_detection_source_ip))
    (n: prot_id=#0 type=16389(nat_detection_destination_ip))
    (n: prot_id=#0 type=16430(status))
06:41:06.133318 IP (tos 0x0, ttl 64, id 14109, offset 0, flags [none], proto UDP (17), length 365)
    178.128.16.**.500 > 151.253.125.**.500: [udp sum ok] isakmp 2.0 msgid 00000000 cookie ca1cc0c13da41457->bd65f2d59661264b: parent_sa ikev2_init[R]:
    (sa: len=44
        (p: #1 protoid=isakmp transform=4 len=44
            (t: #1 type=encr id=aes (type=keylen value=0080))
            (t: #2 type=integ id=#12 )
            (t: #3 type=prf id=#5 )
            (t: #4 type=dh id=modp1024 )))
    (v2ke: len=128 group=modp1024 f47383bee37ab5c042ac033981fc8010586576a7075b1610a41150952a7033523b59c6b07306a1fbabd886894e0d3994c8d64cefdd73c81a8353721acb888fb318e2c84eec627386565cce98f992dc1200e42237dd5005f6985c24bdaf79242c94d37e75b5af1f09e6165878d2eb7623569e58f4e09960d3380fdb87f5103f5d)
    (nonce: len=32 nonce=(1a4864e24db58a5d90879db24ee362c16e9a4cb083e4ad591f4c5a2104cc3ad5) )
    (n: prot_id=#0 type=16388(nat_detection_source_ip))
    (n: prot_id=#0 type=16389(nat_detection_destination_ip))
    (v2cr: len=21 04dfc55921c5256f00589bd2b3cd690f6b159bba27)
    (n: prot_id=#0 type=16404(status))

 iptables -L
Chain INPUT (policy ACCEPT)
target     prot opt source               destination
ACCEPT     all  --  anywhere             anywhere             state RELATED,ESTABLISHED
ACCEPT     tcp  --  anywhere             anywhere             tcp dpt:ssh
ACCEPT     all  --  anywhere             anywhere
ACCEPT     udp  --  anywhere             anywhere             udp dpt:isakmp
ACCEPT     udp  --  anywhere             anywhere             udp dpt:ipsec-nat-t
ACCEPT     tcp  --  anywhere             anywhere             tcp dpt:isakmp
ACCEPT     tcp  --  anywhere             anywhere             tcp dpt:4500
DROP       all  --  anywhere             anywhere

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination
ACCEPT     all  --  172.16.152.116       anywhere             policy match dir in pol ipsec proto esp
ACCEPT     all  --  anywhere             10.130.0.0/16        policy match dir out pol ipsec proto esp
DROP       all  --  anywhere             anywhere

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination

Can anyone look into this issue please.

**Comment by tobiasbrunner** [comment]
Read the logs. Maybe have a look at #196.
